include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -2224,6 +2224,18 @@ namespace internal
             compress_weights(transfer);
         }
     }
+
+
+    template <int dim, typename Number>
+    friend void
+    MGTwoLevelTransferNonNested<dim,
+                                LinearAlgebra::distributed::Vector<Number>>::
+      reinit(const DoFHandler<dim> &          dof_handler_fine,
+             const DoFHandler<dim> &          dof_handler_coarse,
+             const Mapping<dim> &             mapping_fine,
+             const Mapping<dim> &             mapping_coarse,
+             const AffineConstraints<Number> &constraint_fine,
+             const AffineConstraints<Number> &constraint_coarse);
```

Comment:
? I guess this should be in the .h file?

Reply:
The issue is that `MGTwoLevelTransferImplementation::create_coarse_partitioner()` is defined in .templates.h and is private. Some compilers are also not happy with the present PR, so I moved it to public members and removed the friend declaration. Do you agree with that?